vcf_zarr_spec.md

```diff
@@ -0,0 +1,122 @@
+
+(*Implementation note: In general it is not possible to determine the size of some dimensions without a full pass through the VCF file being converted. Implementations may choose to fix the maximum size of some dimensions, although they should warn the user if this results in a lossy conversion.*)
+
+### Fixed fields
+
+The fixed VCF fields `CHROM`, `POS`, `ID`, `REF`, `ALT`, `QUAL`, and `FILTER` are stored as Zarr arrays according to the following table. Note that the `REF` and `ALT` fields are combined into a single Zarr array.
+
+| VCF field(s) | Zarr array path    | Shape                 | Dimension names       | Dtype   |
+|--------------|--------------------|-----------------------|-----------------------|---------|
+| `CHROM`      | `variant_contig`   | `(variants)`          | `[variants]`          | `int`   |
+| `POS`        | `variant_position` | `(variants)`          | `[variants]`          | `int`   |
+| `ID`         | `variant_id`       | `(variants)`          | `[variants]`          | `str`   |
+| `REF`, `ALT` | `variant_allele`   | `(variants, alleles)` | `[variants, alleles]` | `str`   |
+| `QUAL`       | `variant_quality`  | `(variants)`          | `[variants]`          | `float` |
+| `FILTER`     | `variant_filter`   | `(variants, filters)` | `[variants, filters]` | `bool`  |
+
+Each value in the `variant_contig` array is an integer offset into the `contigs` attribute list.
+
+The `variant_filter` array contains a true value at position `i` if the filter at position `i` in the `filters` attribute list applies for a given variant. If not filters have been applied
+for a variant, all values are false.
+
+### INFO fields
+
+Each INFO field is stored as a two-dimensional Zarr array at a path with name `variant_<field>`, of shape `(variants, <Number>)`, dimension names `[variants, <Number dimension name>]`. Dtypes, and missing and fill values are encoded as described above.
+
+### FORMAT fields
+
+Each FORMAT field is stored as a three-dimensional Zarr array at a path with name `call_<field>`, of shape `(variants, samples, <Number>)`, dimension names `[variants, samples, <Number dimension name>]`. Dtypes, and missing and fill values are encoded as described above.
+
+A **Genotype (GT) field** is stored as a three-dimensional Zarr array at a path with name `call_genotype`, of shape `(variants, samples, ploidy)`, with an `int` dtype. Values encode the allele, with 0 for REF, 1 for the first alternate non-reference allele, and so on. A value of -1 indicates missing, and -2 indicates fill in mixed-ploidy datasets.
+
+To indicate phasing, there is an optional accompanying Zarr array at a path with name `call_genotype_phased`, of shape `(variants, samples)`, with dtype `bool`. Values are true if a call is phased, false if unphased (or not present). If the array is not present then all calls are unphased.
```

Comment:
Seems wasteful to force a whole other matrix to say "everthing is phased". I've opened #2 to track.

Reply:
Good point. Implementing this will take a bit of care - see discussion in #2. Are you happy to merge this PR in the meantime?

Reply:
Sure, merge away.